src/layup/utilities/layupConfigs.py

```diff
@@ -0,0 +1,284 @@
+    orientation_constants_url: str = "https://naif.jpl.nasa.gov/pub/naif/generic_kernels/pck/pck00010.tpc"
+    """url of observatory_constants"""
+
+    data_file_list: list = None
+    """convenience list of all the file names"""
+
+    urls: dict = None
+    """dictionary of filename: url"""
+
+    data_files_to_download: list = None
+    """list of files that need to be downloaded"""
+
+    ordered_kernel_files: list = None
+    """list of kernels ordered from least to most precise - used to assemble meta_kernel file"""
+
+    registry: list = None
+    """Default Pooch registry to define which files will be tracked and retrievable"""
+
+    @property
+    def default_url(self):
+        """returns a dictionary of the default urls used in this version of layup"""
+        return {
+            "planet_ephemeris": self.__class__.planet_ephemeris_url,
+            "earth_predict": self.__class__.earth_predict_url,
+            "earth_historical": self.__class__.earth_historical_url,
+            "earth_high_precision": self.__class__.earth_high_precision_url,
+            "jpl_planets": self.__class__.jpl_planets_url,
+            "jpl_small_bodies": self.__class__.jpl_small_bodies_url,
+            "leap_seconds": self.__class__.leap_seconds_url,
+            "observatory_codes": self.__class__.observatory_codes_url,
+            "orientation_constants": self.__class__.orientation_constants_url,
+        }
+
+    @property
+    def default_filenames(self):
+        """returns a dictionary of the default filenames used in this version of layup"""
+        return {
+            "planet_ephemeris": self.__class__.planet_ephemeris,
+            "earth_predict": self.__class__.earth_predict,
+            "earth_historical": self.__class__.earth_historical,
+            "earth_high_precision": self.__class__.earth_high_precision,
+            "jpl_planets": self.__class__.jpl_planets,
+            "jpl_small_bodies": self.__class__.jpl_small_bodies,
+            "leap_seconds": self.__class__.leap_seconds,
+            "meta_kernel": self.__class__.meta_kernel,
+            "observatory_codes": self.__class__.observatory_codes,
+            "orientation_constants": self.__class__.orientation_constants,
+        }
+
+    def __post_init__(self):
+        """Automagically validates the auxiliary configs after initialisation."""
+        self._create_lists_auxiliary_configs()
+        self._validate_auxiliary_configs()
+
+    def _validate_auxiliary_configs(self):
+        """
+        validates the auxililary config attributes after initialisation.
+        """
+        for file in self.default_filenames:
+            if file != "meta_kernel":
+                if (
+                    self.default_filenames[file] == getattr(self, file)
+                    and getattr(self, file + "_url") != self.default_url[file]
+                ):
+                    logging.error(f"ERROR: url for {file} given but filename for {file} not given")
```

Comment:
Current consensus is that we don't want logging, so everywhere there's logging should be printing to stderror or stdout